### PR TITLE
Add String Session support and Deploy to Heroku button

### PR DESCRIPTION
**Pull request by DragonightFury**

### CODE REVIEW

**Comment by AvinashReddy3108** [comment]
Glad to see someone porting my "Extended" stuff into the original project.

**Comment by zakaryan2004** [comment]
Is this abandoned? Doesn't seem to be complete per what you told me.

**Comment by AvinashReddy3108** [comment]
> Is this abandoned? Doesn't seem to be complete per what you told me.

Looks pretty complete to me, needs the wiki to be updated if we merge this tho.

**Comment by DragonightFury** [comment]
Why from Extended? Look into other bots. HOW could i do that in the other way?

File: README.md [hunk 1]
```diff
@@ -31,6 +31,9 @@ It has since evolved, becoming extremely modular and simple to use.
 For configuring Paperplane, you can checkout our [Wiki](https://wiki.raphielgang.org). Please, before asking us in our groups,
 make sure you have followed the Wiki and all the steps required.
 
+The easiest way to deploy this usebot:
```
Comment (zakaryan2004) [inline README.md:34]:
```suggestion
Deploy Paperplane to Heroku:
```

File: generate_string_session.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
+from telethon.sync import TelegramClient
+from telethon.sessions import StringSession
+
+print("""Please go-to my.telegram.org
```
Comment (zakaryan2004) [inline generate_string_session.py:4]:
```suggestion
print("""Please go to my.telegram.org
```

File: generate_string_session.py [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+Login using your Telegram account
+Click on API Development Tools
+Create a new application, by entering the required details""")
+
+API_KEY = input("API_KEY: ")
+API_HASH = input("API_HASH: ")
+
+with TelegramClient(StringSession(), API_KEY, API_HASH) as client:
+    print("Here is your userbot srting, copy it to config.env!")
```
Comment (zakaryan2004) [inline generate_string_session.py:13]:
```suggestion
    print("This is your string session, be careful with it and don't share it with anyone else!")
```

**Comment by AvinashReddy3108** [comment]
> Why from Extended? Look into other bots. HOW could i do that in the other way?

Other ones are clones of mine.

**Comment by RealAkito** [comment]
STRING_SESSION is surely dangerous though, that's my opinion.

**Comment by zakaryan2004** [comment]
> STRING_SESSION is surely dangerous though, that's my opinion.

It really is, easier to leak, especially when Heroku doesn't encrypt its env vars. But well, it's one thing which will drastically make Heroku deployment easier, and also if you think about it being dangerous, then userbot.session and even other env vars can be considered dangerous too. If one can't treat the String Session as password, then they pretty much shouldn't use Paperplane or anything else on the Internet.

**Comment by zakaryan2004** [comment]
> > Why from Extended? Look into other bots. HOW could i do that in the other way?
> 
> Other ones are clones of mine.

If you have any authorship issues, feel free to write.

File: app.json [hunk 4]
```diff
@@ -44,23 +44,27 @@
         },
         "OPEN_WEATHER_MAP_APPID": {
             "description": "OpenWeather Map API Key for .weather command. Get from https://openweathermap.org/.",
-            "required": true 
+            "required": false
         },
         "BOTLOG": {
             "description": "In case you want to turn on logging, set this True.",
             "value": "False"
```
Comment (zakaryan2004) [inline app.json:51]:
```suggestion
            "value": "False",
```

File: app.json [hunk 5]
```diff
@@ -44,23 +44,27 @@
...
+            "required": false 
         },
         "BOTLOG_CHATID": {
             "description": "Chat ID for Botlog. Get it from group management bots such as Marie or her forks.",
             "value": "0"
```
Comment (zakaryan2004) [inline app.json:56]:
```suggestion
            "value": "0",
```

File: app.json [hunk 6]
```diff
@@ -44,23 +44,27 @@
...
+            "required": false 
         },
         "CONSOLE_LOGGER_VERBOSE": {
             "description": "If you want verbosity on the Logging.",
             "value": "False"
```
Comment (zakaryan2004) [inline app.json:61]:
```suggestion
            "value": "False",
```

File: app.json [hunk 7]
```diff
@@ -44,23 +44,27 @@
...
+            "required": false 
         },
         "PM_AUTO_BAN": {
             "description": "PM Auto-Ban Feature Switch.",
             "value": "False"
```
Comment (zakaryan2004) [inline app.json:66]:
```suggestion
            "value": "False",
```

**Comment by zakaryan2004** [comment]
Please check if everything is working as intended.

**Comment by RealAkito** [comment]
Did heroku not use 

 "value": false,

instead of

 "value": "False",

?

**Pull request merged**